streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBuffer.java

```diff
@@ -0,0 +1,185 @@
+                    }
+                    return;
+                }
+
+                BufferValue bufferValue = BufferValue.deserialize(ByteBuffer.wrap(keyValue.value));
+                K key = keySerde.deserializer().deserialize(topic,
+                    PrefixedWindowKeySchemas.TimeFirstWindowKeySchema.extractStoreKeyBytes(keyValue.key.get()));
+
+                Change<V> value = valueSerde.deserializeParts(
+                    topic,
+                    new Change<>(bufferValue.newValue(), bufferValue.oldValue())
+                );
+                while (keyValue != null && predicate.get()) {
+                    if (bufferValue.context().timestamp() != minTimestamp) {
+                        throw new IllegalStateException(
+                            "minTimestamp [" + minTimestamp + "] did not match the actual min timestamp [" +
+                                bufferValue.context().timestamp() + "]"
+                        );
+                    }
+                    callback.accept(new Eviction<>(key, value, bufferValue.context()));
+                    wrapped().remove(keyValue.key);
+                    numRecords--;
+                    bufferSize = bufferSize - computeRecordSize(keyValue.key, bufferValue);
+                    if (iterator.hasNext()) {
+                        keyValue = iterator.next();
+                        bufferValue = BufferValue.deserialize(ByteBuffer.wrap(keyValue.value));
+                        key = keySerde.deserializer().deserialize(topic,
+                            PrefixedWindowKeySchemas.TimeFirstWindowKeySchema.extractStoreKeyBytes(keyValue.key.get()));
+                        value = valueSerde.deserializeParts(
+                            topic,
+                            new Change<>(bufferValue.newValue(), bufferValue.oldValue())
+                        );
+                        minTimestamp = bufferValue.context().timestamp();
+
+                    } else {
+                        keyValue = null;
+                        minTimestamp = Long.MAX_VALUE;
+                    }
+                }
+            }
+        }
+    }
+
+
+    @Override
+    public Maybe<ValueAndTimestamp<V>> priorValueForBuffered(final K key) {
+        return Maybe.undefined();
+    }
+
+    @Override
+    public void put(final long time, final Record<K, Change<V>> record, final ProcessorRecordContext recordContext) {
+        requireNonNull(record.value(), "value cannot be null");
+        requireNonNull(recordContext, "recordContext cannot be null");
+        if (wrapped().observedStreamTime - gracePeriod > record.timestamp()) {
+            return;
+        }
+        final Bytes serializedKey = Bytes.wrap(
+            PrefixedWindowKeySchemas.TimeFirstWindowKeySchema.toStoreKeyBinary(keySerde.serializer().serialize(topic, record.key()),
+                record.timestamp(),
+                Long.valueOf(recordContext.offset()).intValue()).get());
```

Comment:
Isn't this the same as this?
```suggestion
                (int) recordContext.offset()).get());
```

Reply:
For some reason I thought there was overflow protection, but apparently not